addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -0,0 +1,864 @@
+
+    l10n_ro_edi_stock_end_bcp = fields.Selection(selection=BORDER_CROSSING_POINTS, string="Border Crossing Point")
+    l10n_ro_edi_stock_end_customs_office = fields.Selection(selection=CUSTOMS_OFFICES, string="Customs Office")
+    l10n_ro_edi_stock_end_state_id = fields.Many2one(comodel_name='res.country.state', string="State", domain="[('country_id.code', '=', 'RO')]")
+    l10n_ro_edi_stock_end_city = fields.Char(string="City")
+    l10n_ro_edi_stock_end_street = fields.Char(string="Street")
+    l10n_ro_edi_stock_end_zip = fields.Char(string="Postal Code")
+    l10n_ro_edi_stock_end_other_info = fields.Char(string="Other Info")
+
+    l10n_ro_edi_stock_remarks = fields.Text(string="Remarks")
+
+    # View control fields
+    l10n_ro_edi_stock_enable = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable')
+    l10n_ro_edi_stock_enable_send = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_send')
+    l10n_ro_edi_stock_enable_fetch = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_fetch')
+    l10n_ro_edi_stock_enable_send_info = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_send_info')
+    l10n_ro_edi_stock_enable_amend = fields.Boolean(compute='_compute_l10n_ro_edi_stock_enable_amend')
+
+    l10n_ro_edi_stock_fields_readonly = fields.Boolean(compute='_compute_l10n_ro_edi_stock_fields_readonly')
+
+    ################################################################################
+    # Reset Data Methods
+    ################################################################################
+
+    @api.onchange('l10n_ro_edi_stock_operation_type')
+    def _l10n_ro_edi_stock_reset_location_type(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_start_loc_type = 'location'
+            picking.l10n_ro_edi_stock_end_loc_type = 'location'
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_start_loc_type')
+    def _l10n_ro_edi_stock_reset_start_location_data(self):
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('start')
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_end_loc_type')
+    def _l10n_ro_edi_stock_reset_end_location_data(self):
+        for picking in self:
+            picking._l10n_ro_edi_stock_reset_location_data('end')
+
+    def _l10n_ro_edi_stock_reset_location_data(self, location: str):
+        for picking in self:
+            picking[f'l10n_ro_edi_stock_{location}_bcp'] = False
+            picking[f'l10n_ro_edi_stock_{location}_customs_office'] = False
+            picking[f'l10n_ro_edi_stock_{location}_state_id'] = False
+            picking[f'l10n_ro_edi_stock_{location}_zip'] = False
+            picking[f'l10n_ro_edi_stock_{location}_city'] = False
+            picking[f'l10n_ro_edi_stock_{location}_street'] = False
+            picking[f'l10n_ro_edi_stock_{location}_other_info'] = False
```

Comment:
Usually we don't like much adding onchanges. Also, I'm not sure why you have to reset everything here. Suppose the partner made a mistake and forget to change the stock operation type. It could be frustrating to lost the all address you just made